packages/plugin-core/src/views/DendronTreeViewV2.ts

```diff
@@ -9,41 +9,57 @@ import {
   VSCodeEvents,
 } from "@dendronhq/common-all";
 import { getDurationMilliseconds } from "@dendronhq/common-server";
-import { WorkspaceUtils } from "@dendronhq/engine-server";
+import { EngineEventEmitter, WorkspaceUtils } from "@dendronhq/engine-server";
 import _ from "lodash";
 import path from "path";
 import * as vscode from "vscode";
+import { Disposable } from "vscode-languageclient";
 import { GotoNoteCommand } from "../commands/GotoNote";
 import { IDendronExtension } from "../dendronExtensionInterface";
 import { Logger } from "../logger";
 import { AnalyticsUtils } from "../utils/analytics";
 import { VSCodeUtils } from "../vsCodeUtils";
-import { WSUtils } from "../WSUtils";
 import { WebViewUtils } from "./utils";
 
 /**
  * Class managing the webview version of the Dendron tree view (currently called
  * TreeViewV2) - this is the side panel UI that gives the webview/react/antd
  * based tree view of the Dendron note hierarchy
  */
-export class DendronTreeViewV2 implements vscode.WebviewViewProvider {
+export class DendronTreeViewV2
+  implements vscode.WebviewViewProvider, Disposable
+{
   public static readonly viewType = DendronTreeViewKey.TREE_VIEW_V2;
 
   private _view?: vscode.WebviewView;
   private _ext: IDendronExtension;
+  private _onEngineNoteStateChangedDisposable: Disposable | undefined;
+  private _engineEvents;
 
-  constructor(ext: IDendronExtension) {
+  /**
+   *
+   * @param engineEvents - specifies when note state has been changed on the
+   * engine
+   */
+  constructor(ext: IDendronExtension, engineEvents: EngineEventEmitter) {
     this._ext = ext;
+    this._engineEvents = engineEvents;
 
     this._ext.context.subscriptions.push(
       vscode.window.onDidChangeActiveTextEditor(this.onOpenTextDocument, this)
     );
 
-    this._ext.noteSyncService.onNoteChange((noteProps) => {
-      if (this._view && this._view.visible) {
-        this.refresh(noteProps);
-      }
-    });
+    this._onEngineNoteStateChangedDisposable =
+      this._engineEvents.onEngineNoteStateChanged((noteChangeEntry) => {
+        const ctx = "refreshDendronTreeViewV2EngineNoteStateChanged";
+        Logger.info({ ctx });
+        noteChangeEntry.map((changeEntry) => this.refresh(changeEntry.note));
+      });
+  }
+  dispose(): void {
```

Comment:
I don't think we've hooked up the calling of this `dispose` in workspace.ts yet when we construct the tree view?

Reply:
ah good catch